data-prepper-plugins/peer-forwarder/src/main/java/com/amazon/dataprepper/plugins/prepper/peerforwarder/discovery/AwsCloudMapPeerListProvider.java

```diff
@@ -106,6 +113,16 @@ private static String getRequiredSettingString(final PluginSetting pluginSetting
         return Objects.requireNonNull(propertyValue, String.format("Missing '%s' configuration value", propertyName));
     }
 
+    private static Map<String, String> getOptionalSettingMap(final PluginSetting pluginSetting, final String propertyName) {
+        final Map<String, String> propertyValue = pluginSetting.getTypedMap(propertyName, String.class, String.class);
+
+        if (propertyValue == null) {
```

Comment:
Out of scope for PR: We should eliminate null checks by enforcing a default value in the PluginSettings or allow calling classes to specify the default. Raising: #1563